src/library/library.cpp

```diff
@@ -569,27 +570,42 @@ void Library::setEditMedatataSelectedClick(bool enabled) {
     emit(setSelectedClick(enabled));
 }
 
+bool Library::saveTrack(const TrackPointer& pTrack) {
+    VERIFY_OR_DEBUG_ASSERT(pTrack) {
+        return false;
+    }
+    if (!pTrack->isDirty()) {
+        return false;
+    }
+    saveTrack(pTrack.get(), TrackMetadataExportMode::Deferred);
+    DEBUG_ASSERT(!pTrack->isDirty());
+    return true;
+}
+
+// Export metadata and save the track in both the internal database
+// and external libaries.
 void Library::saveEvictedTrack(Track* pTrack) noexcept {
-    // It can produce dangerous signal loops if the track is still
-    // sending signals while being saved!
-    // See: https://bugs.launchpad.net/mixxx/+bug/1365708
```

Comment:
The disconnecting code is part of GlobalTrackCache::saveEvictedTrack(). I have amended it with a more appropriate description regarding _objects-under-destruction_.

I don't experience any issues while saving intermediate modifications in the internal and external collections. Otherwise this would be a design or programming error that needs to be fixed.